File: benchmarks/pandas/bench_countna.py
"""Benchmark: countna — count NaN/null values in a Series with 10% nulls"""
import json, time
import pandas as pd
import numpy as np

ROWS = 100_000
WARMUP = 3
ITERATIONS = 20

data = [None if i % 10 == 0 else float(i) for i in range(ROWS)]
s = pd.Series(data)

for _ in range(WARMUP):
    s.isna().sum()

start = time.perf_counter()
for _ in range(ITERATIONS):
    s.isna().sum()
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "countna",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))

File: benchmarks/tsb/bench_countna.ts
/**
 * Benchmark: countna — count NaN/null values in a Series with 10% nulls
 */
import { Series } from "../../src/index.js";
import { countna } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 20;

const data = Array.from({ length: ROWS }, (_, i) => (i % 10 === 0 ? null : i));
const s = new Series({ data });

for (let i = 0; i < WARMUP; i++) {
  countna(s);
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  countna(s);
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "countna",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);
